docs/datafeed.md

```diff
@@ -0,0 +1,136 @@
+```
+
+The minimal configuration for the datafeed service is the version of the datafeed which will be chosen to be use in the bdk.
+If the bot developers want to use the datafeed version 2 in their bot, the version configuration have to be specified as `v2`.
+Otherwise, the datafeed version 1 will be used by default.
+
+Bot developers can also configure a dedicated retry mechanism which will be used only by the datafeed service.
+Basically, the datafeed service retry configuration has the field same as the global retry configuration with the fields for implementing 
+the exponential backoff mechanism.
+
+## Subscribe/Unsubscribe RealTimeEventListener
+
+[RealTimeEventListener](https://javadoc.io/doc/com.symphony.platformsolutions/symphony-bdk-core/latest/com/symphony/bdk/core/service/datafeed/RealTimeEventListener.html) is an interface definition for a callback to be invoked when a real-time event is received from the datafeed.
+This real-time event can be one of these following event types:
+
+- Message Sent
+- Messages Suppressed
+- Symphony Elements Action
+- Shared Wall Post
+- IM/MIM Created
+- Room Created
+- Room Updated Message
+- Room Deactivated Message
+- Room Reactivated Message
+- User Requested to Join Room
+- User Joined Room
+- User Left Room
+- Room Member Promoted To Owner
+- Room Member Demoted From Owner
+- Connection Requested
+- Connection Accepted
+
+The Datafeed Service can subscribe/unsubscribe one or many `RealTimeEventListener` by calling `DatafeedService#subscribe` or
+`DatafeedService#unsubscribe`. For instance:
+
+```java
+public class Example {
+    
+    public static void main(String[] args) {
+        final SymphonyBdk bdk = new SymphonyBdk(loadFromClasspath("/config.yaml"));
+        final RealTimeEventListener listener = new RealTimeEventListener() {
+        @Override
+            public void onMessageSent(V4Initiator initiator, V4MessageSent event) {
+              log.info("Message sent");
+            }
+        };
+        // subscribe a listener
+        bdk.datafeed().subscribe(listener);
+        // unsubscribe a listener
+        bdk.datafeed().unsubscribe(listener);
+        // start the datafeed service
+        bdk.datafeed().start();
+    }
+}
+```
+
+## Read a datafeed
+
+After subscribe a `RealTimeEventListener`, a bot can start the datafeed service by calling `DatafeedService#start()`.
+This method will make the bot starting reading the `RealTimeEventListener` from the datafeed. Once the datafeed is started,
+if bot developers call this method again, it will throw a `IllegalStateException` saying that the reading loop has started already.
+
+Bot developers can also stop the Datafeed service anytime by calling the method `DatafeedService#stop()`.
+After this method is called, if there is still any request from the bot to the datafeed, the bot will finish this request before really stop the Datafeed service.
+
+For instance:
+```java
+public class Example {
```

Comment:
Isn't the example redundant with the how to use at the beginning? perhaps we can keep only one of it?